Assignment4/haolin.chen-E94054014/task06.py

```diff
@@ -0,0 +1,94 @@
+# -*- coding: utf-8 -*-
+"""Task06.ipynb
+
+Automatically generated by Colaboratory.
+
+Original file is located at
+    https://colab.research.google.com/drive/1cgsSvhue5OhhPP0j87kpUgo3CZRvquPp
+
+**Task 06: Modifying RDF(s)**
+"""
+
+!pip install rdflib
+github_storage = "https://raw.githubusercontent.com/FacultadInformatica-LinkedData/Curso2023-2024/master/Assignment4/course_materials"
+
+"""Read the RDF file as shown in class"""
+
+from rdflib import Graph, Namespace, Literal
+from rdflib.namespace import RDF, RDFS
+g = Graph()
+g.namespace_manager.bind('ns', Namespace("http://somewhere#"), override=False)
+g.namespace_manager.bind('vcard', Namespace("http://www.w3.org/2001/vcard-rdf/3.0#"), override=False)
+g.parse(github_storage+"/rdf/example5.rdf", format="xml")
+
+"""Create a new class named Researcher"""
+
+ns = Namespace("http://somewhere#")
+g.add((ns.Researcher, RDF.type, RDFS.Class))
+for s, p, o in g:
+  print(s,p,o)
+
+"""**TASK 6.1: Create a new class named "University"**
+
+"""
+
+# TO DO
+g.add((ns.University, RDF.type, RDFS.Class))
+
+# Visualize the results
+for s, p, o in g:
+  print(s,p,o)
+
+"""**TASK 6.2: Add "Researcher" as a subclass of "Person"**"""
+
+# TO DO
+g.add((ns.Researcher, RDFS.subClassOf, ns.Person))
+
+# Visualize the results
+for s, p, o in g:
+  print(s,p,o)
+
+"""**TASK 6.3: Create a new individual of Researcher named "Jane Smith"**"""
+
+# TO DO
+g.add((ns.JaneSmith, RDF.type, ns.Researcher))
+
+# Visualize the results
+for s, p, o in g:
+  print(s,p,o)
+
+"""**TASK 6.4: Add to the individual JaneSmith the email address, fullName, given and family names**"""
+
+# TO DO
+vcard = Namespace("http://www.w3.org/2001/vcard-rdf/3.0#")
+
+# Adding info to JaneSmith
+g.add((ns.JaneSmith, vcard.Email, Literal("jane.smith@email.com")))
+g.add((ns.JaneSmith, vcard.FN, Literal("Jane Smith")))
+g.add((ns.JaneSmith, vcard.Given, Literal("Jane")))
+g.add((ns.JaneSmith, vcard.Family, Literal("Smith")))
+
+# Visualize the results
+for s, p, o in g:
+    print(s, p, o)
+
+"""**TASK 6.5: Add UPM as the university where John Smith works**"""
+
+# TO DO
+g.add((ns.UPM, RDF.type, ns.University))
+g.add((ns.JohnSmith, vcard.WorkFor, ns.UPM))
```

Comment:
vcard does not define WorkFor. Create it using your own namespace

Reply:
# updated TASK 6.5
g.add((ns.UPM, RDF.type, ns.University))
g.add((ns.JohnSmith, ns.WorkFor, ns.UPM))